File: src/test/java/com/snackgame/server/member/domain/GuestTest.java
package com.snackgame.server.member.domain;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

@SuppressWarnings("NonAsciiCharacters")
@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class GuestTest {

    @Test
    void 이름앞에_접두사가_붙는다() {
        var guest = new Guest();

        assertThat(guest.getName()).startsWith("게스트#");
    }

    @Test
    void 이름의_8자리를_무작위로_생성한다() {
        var guest = new Guest();

        assertThat(guest.getName().substring(4)).hasSize(8);
    }
}

File: src/test/java/com/snackgame/server/member/domain/GuestTest.kt
@file:Suppress("NonAsciiCharacters")

package com.snackgame.server.member.domain

import com.snackgame.server.member.exception.GuestRestrictedException
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

internal class GuestTest {

    @Test
    fun `게스트는 이름을 변경할 수 없다`() {
        assertThatThrownBy {
            Guest(Name("게스트")).changeNameTo(Name("이름"))
        }.isInstanceOf(GuestRestrictedException::class.java)
    }

    @Test
    fun `게스트는 그룹을 변경할 수 없다`() {
        assertThatThrownBy {
            Guest(Name("게스트")).changeGroupTo(Group("그룹"))
        }.isInstanceOf(GuestRestrictedException::class.java)
    }

    @Test
    fun `게스트의 계정 타입은 GUEST이다`() {
        assertThat(Guest(Name("게스트")).accountType).isEqualTo(AccountType.GUEST)
    }
}
